instrumentation/mongodb/src/main/java/brave/mongodb/MongoDBTracing.java

```diff
@@ -0,0 +1,128 @@
+      "convertToCapped", "create", "createIndexes", "drop", "dropIndexes", "killCursors", "listIndexes", "reIndex"));
+    int maxAbbreviatedCommandLength = 1000;
+
+    Builder(Tracing tracing) {
+      if (tracing == null) throw new NullPointerException("tracing == null");
+      this.tracing = tracing;
+    }
+
+    Builder(MongoDBTracing mongoDBTracing) {
+      this(mongoDBTracing.tracing);
+      maxAbbreviatedCommandLength(mongoDBTracing.maxAbbreviatedCommandLength);
+      clearCommandsWithCollectionName();
+      addAllCommandsWithCollectionName(mongoDBTracing.commandsWithCollectionName);
+    }
+
+    /**
+     * Sets how many characters of the MongoDB command to report in the "mongodb.command" tag. Defaults to 1000.
+     *
+     * If zero, command reporting will be disabled. The command name will still be reported as "mongodb.command.name".
+     *
+     * Set it to a large number (ex. {@link Integer#MAX_VALUE}) to disable truncation.
+     */
+    public Builder maxAbbreviatedCommandLength(int maxAbbreviatedCommandLength) {
+      if (maxAbbreviatedCommandLength < 0) throw new IllegalArgumentException("maxAbbreviatedCommandLength < 0");
+      this.maxAbbreviatedCommandLength = maxAbbreviatedCommandLength;
+      return this;
+    }
+
+    /**
+     * Clear the allow-list of command names for which tracing will attempt to extract the collection/view name from
+     * the argument.
+     */
+    public Builder clearCommandsWithCollectionName() {
+      commandsWithCollectionName.clear();
+      return this;
+    }
+
+    /**
+     * Adds a MongoDB command name to the allow-list, indicating that if the command's argument is a string, the
+     * argument contains the collection/view name. Tracing will only attempt to extract the collection/view name
+     * argument from allow-listed commands.
+     *
+     * The default allow-list is a set of commonly used commands ({@link #commandsWithCollectionName}) that operate on
+     * collections/views.
+     *
+     * @param commandName command name to add to the allow-list
+     */
+    public Builder addCommandWithCollectionName(String commandName) {
+      commandsWithCollectionName.add(commandName);
+      return this;
+    }
+
+    /**
+     * Same as {@link #addCommandWithCollectionName(String)} but for a collection of command names.
+     */
+    public Builder addAllCommandsWithCollectionName(Collection<String> commandNames) {
+      commandsWithCollectionName.addAll(commandNames);
+      return this;
+    }
+
+    public MongoDBTracing build() {
+      return new MongoDBTracing(this);
+    }
+  }
+
+  final Tracing tracing;
+  final int maxAbbreviatedCommandLength;
+  final Set<String> commandsWithCollectionName;
+
+  MongoDBTracing(Builder builder) {
+    tracing = builder.tracing;
+    maxAbbreviatedCommandLength = builder.maxAbbreviatedCommandLength;
+    commandsWithCollectionName = Collections.unmodifiableSet(builder.commandsWithCollectionName);
```

Comment:
Copy into a new set too, the user will still be able to modify the set through the builder.